### PR TITLE
THREESCALE-9183: Removing leftovers of source2swagger

### PR DESCRIPTION
**Pull request by mayorova**
**What this PR does / why we need it**:

Cleaning up old source2swagger annotations and related code.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-9183

**Verification steps** 
-none-

**Special notes for your reviewer**:

Based on https://github.com/3scale/porta/pull/3103

### CODE REVIEW

**Comment by nidhi-soni1104** [comment]
Looks good to me

**Review by nidhi-soni1104 — DISMISSED** [review]
LGTM

**Review by jlledom — APPROVED** [review]
I'm all for this cleanup. If the tests pass, approved. Just one thing: the comments removed here contain a list of accepted parameters for each endpoint which I find useful, shouldn't we keep the list?, no matter the format. 

The idea is to replace source2swagger by rswag in the future. If I'm not wrong, I'd say in rswag you need to write specs where you define the interface for the endpoint, so we'll probably need the list of parameters in order to write those specs.

File: Gemfile [hunk 1]
```diff
@@ -250,8 +250,6 @@ group :development, :test do
   gem 'pry-shell'
   gem 'pry-stack_explorer'
 
-  # for `rake doc:liquid:generate` and similar
```
Comment (josemigallas) [inline Gemfile:253]: Is this relevant? Should we remove/edit this task?

Comment (mayorova) [reply Gemfile:253]: I think this was a "typo" and it referred to `doc:swagger:generate` which I already removed.

Comment (akostadinov) [reply Gemfile:253]: Do we still need `swagger-ui_rails`?

File: app/lib/logic/rolling_updates.rb [hunk 2]
```diff
@@ -118,12 +118,6 @@ def missing_config
         end
       end
 
-      class NewProviderDocumentation < Base
-        def missing_config
-          false
-        end
-      end
```
Comment (akostadinov) [inline app/lib/logic/rolling_updates.rb:125]: Apparently this leftover was not used anymore. But what if old customers have this rolling update in their custom configuration? Would then their upgrade fail? Should we keep this forever or document that it needs to be removed before upgrade?

File: config/routes.rb [hunk 3]
```diff
@@ -299,7 +299,6 @@ def call(env)
 
       namespace :api_docs do
         resource :account_data, :only => [:show]
-        resources :specs, only: :show
```
Comment (akostadinov) [inline config/routes.rb:302]: what was that?

**Review by akostadinov — APPROVED** [review]
Some questions left on your consideration. lgtm, thanks!

**Review by lvillen — APPROVED** [review]

File: Gemfile (hunk 1 above)
Comment (mayorova) [reply Gemfile:253]: > Do we still need `swagger-ui_rails`?

I'm afraid we do. I believe this is for Swagger v1, and we still support rendering Swagger UI for v1 specs for our customers.

File: app/lib/logic/rolling_updates.rb (hunk 2 above)
Comment (mayorova) [reply app/lib/logic/rolling_updates.rb:125]: You mean if they have `new_provider_documentation: true` in their `rolling_updates.yml`.

Well, they are not supposed to touch this configuration in the first place :)

Secondly, why would they do that? Just to load a broken Swagger v2 Active Docs for 3scale APIs? :)

And most importantly, the `rolling_updates.yml` can contain extra features, and if there is no code to process them, then they would be just ignored.

Comment (akostadinov) [reply app/lib/logic/rolling_updates.rb:125]: hmm, I think it failed for me like that in the past, maybe don't remember. But it makes sense, nobody would have used v2.

**Pull request merged**